tasks/Module 4/task_2.py

```diff
@@ -0,0 +1,32 @@
+"""Task 2 of Module 4."""
+from pathlib import Path
+import os
+from tkinter.filedialog import askdirectory
+
+
+def filling_in_the_gaps():
+    """
+    Find all files with a given prefix,
+    in a single folder and fill in any gaps in the numbering.
+    """
+    dir_path = askdirectory(title='Select Folder with spam* files')
+    iter = 1
+
+    os.chdir(dir_path)
+
+    for file in sorted(os.listdir(dir_path)):
+        file_path = Path(file)
+
+        files_number = file_path.stem[4:7]
+        file_prefix = file_path.stem[:4]
+
+        expected_number = str(iter).zfill(3)
+
+        if files_number != expected_number:
+            os.rename(file, file_prefix + expected_number + ".txt")
+            iter += 1
+        else:
```

Comment:
Two points:
- You are incrementing `iter` in both cases, so it can be pulled outside the `if...else` statement. In fact you only need `if` part.
- Python can handle it for you :) If you have a for loop and a variable that is incremented in each step, then use `enumerate` instead:
```python
for index, file in enumerate(sorted(os.listdir(dir_path)), start=1):
```
Please noticed that I renamed `iter` to `index` in the example, because `iter` is name of built-in function.

Reply:
`else` removed

`unemerate` used in for loop